parsely/src/main/java/com/parsely/parselyandroid/ParselyAPIConnection.kt

```diff
@@ -0,0 +1,62 @@
+/*
+    Copyright 2016 Parse.ly, Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+package com.parsely.parselyandroid
+
+import android.os.AsyncTask
+import java.net.HttpURLConnection
+import java.net.URL
+
+@Suppress("DEPRECATION")
+internal class ParselyAPIConnection(private val tracker: ParselyTracker) : AsyncTask<String?, Exception?, Void?>() {
+    private var exception: Exception? = null
+
+    @Deprecated("Deprecated in Java")
+    override fun doInBackground(vararg data: String?): Void? {
+        var connection: HttpURLConnection? = null
```

Comment:
Minor (🔍): Consider updating this local variable to a non-null `val`?

`val connection: HttpURLConnection`

PS: This local variable can also be moved within the `try/catch` block it seems, which IMHO is better to have it as close to usage, right? 🤔

Reply:
Good point - I could make it more readable but at the same point it feels okay for now, and very soon it'll be rewritten (coroutines) so I'll address your comment there then 👍